### PR TITLE
feat: infer ops search_path from schema refs

### LINKED ISSUE #439: feat(ops): infer search_path from IR schema references
## Work Type
`product`

## Hill Supported
An operator can generate hardened ops SQL without needing to hand-specify `--ops-search-path` for the common schema-aware case.

## Sponsor Actor
- Operator generating ops SQL from IR/SDL
- Maintainer hardening ops defaults without extra user ceremony

## Playback
A user generates ops SQL from an IR with schema references and gets a safe default `search_path` automatically, while documented overrides still work for edge cases.

## Problem
Wesley now emits explicit `search_path` when asked, but the common path still depends on manual `--ops-search-path` input. That leaves safe defaults underpowered for multi-schema or hardened deployments.

## Proposed Change
- infer a safe default `SET search_path` from IR or schema references
- preserve documented manual override semantics
- cover multi-schema and wrong-schema edge cases with regressions

## Invariants
- emitted ops SQL stays explicit about search-path behavior
- manual overrides remain available and documented
- generated defaults favor deterministic, hardened name resolution

## Non-Goals
- silently widening search paths for convenience
- removing manual override support
- treating ambient database defaults as sufficient

## Acceptance / Tests
- emitted ops path derives a safe default search path from IR or schema references
- manual override semantics are documented
- regression tests cover multi-schema and wrong-schema edge cases

### PR DESCRIPTION
**Pull request by flyingrobots**
## Summary
- infer hardened default ops `search_path` values from IR metadata and schema-qualified plan refs
- preserve manual `--ops-search-path` overrides while keeping schema-aware base table qualification
- document the new default and cover the common, override, and multi-schema cases with regressions

## Testing
- node --test packages/wesley-core/test/unit/qir-translate-env.test.mjs packages/wesley-core/test/snapshots/qir-emission.test.mjs packages/wesley-cli/test/generate-ops.test.mjs
- pnpm lint
- node scripts/check-doc-truth.mjs
- node scripts/check-doc-links.mjs
- node scripts/pre-push-sanity.mjs --files $(git show --pretty='' --name-only HEAD)

Closes #439

### CODE REVIEW

File: packages/wesley-cli/src/commands/generate-ops.mjs [hunk 1]
```diff
@@ -405,7 +408,170 @@ function normalizeOpsSearchPath(setSearchPath, normalizedSchema) {
...
+  for (const schemaName of planSchemaRefs) {
+    const normalized = sanitizeSchemaName(schemaName, normalizedSchema);
+    if (normalized) inferred.add(normalized);
+  }
+
+  const inferredSchemas = Array.from(inferred).sort();
+  const hasQualifiedTableRefs = planSchemaRefs.size > 0;
+  const baseSchema = hasQualifiedTableRefs
+    ? null
+    : inferredSchemas.length === 1
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply packages/wesley-cli/src/commands/generate-ops.mjs]: Fixed on the branch in `56322dc`, then hardened again in `d9099d1`.

`inferOpsSchemaContext()` now keeps the fallback base schema when unqualified refs coexist with qualified refs instead of nulling it for the whole batch. The regression coverage is in `packages/wesley-cli/test/generate-ops.test.mjs:105-158`, including both the mixed join case and the mixed batch case (`baseSchema` stays `app` when IR provides it, and falls back to `public` when it does not).`
